### PR TITLE
Added 0.59.0 release notes

### PR DESCRIPTION
**Pull request by kc611**
As titled.

### CODE REVIEW

**Review by gmarkall — APPROVED** [review]
The notes look good to me.

File: docs/source/release/0.59.0-notes.rst [hunk 1]
```diff
@@ -0,0 +1,527 @@
...
+* PR `#9331 <https://github.com/numba/numba/pull/9331>`_: Fix Syntax and Deprecation Warnings from 3.12. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#9334 <https://github.com/numba/numba/pull/9334>`_: Fix parfor reduction issue with Python 3.12. (`DrTodd13 <https://github.com/DrTodd13>`_)
+* PR `#9335 <https://github.com/numba/numba/pull/9335>`_: Add validation capability for user generated towncrier .rst files. (`kc611 <https://github.com/kc611>`_)
+* PR `#9336 <https://github.com/numba/numba/pull/9336>`_: Add TargetLibraryInfo pass to CPU LLVM pipeline. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#9337 <https://github.com/numba/numba/pull/9337>`_: Revert #8583 which skip tests due to M1 RuntimeDyLd Assertion error (`sklam <https://github.com/sklam>`_)
+* PR `#9341 <https://github.com/numba/numba/pull/9341>`_: Add configuration variable to force llvmlite memory manager on / off (`gmarkall <https://github.com/gmarkall>`_)
+* PR `#9342 <https://github.com/numba/numba/pull/9342>`_: Fix flake8 checks for v6.1.0 (`gmarkall <https://github.com/gmarkall>`_)
```
Comment (stuartarchibald) [inline docs/source/release/0.59.0-notes.rst:505]: Was this generated against current HEAD of `main` (48fe645)? PRs merged recently, e.g. #9360 are missing?

File: docs/source/release/0.59.0-notes.rst [hunk 2]
```diff
@@ -0,0 +1,527 @@
+
+Version 0.59.0 (Release Date)
```
Comment (stuartarchibald) [inline docs/source/release/0.59.0-notes.rst:2]: Reminder to fix the release date. Will make this a merge blocking comment.

**Review by stuartarchibald — CHANGES_REQUESTED** [review]

File: docs/source/release/0.59.0-notes.rst (hunk 2 above)
Comment (kc611) [inline docs/source/release/0.59.0-notes.rst:2]:
```suggestion
Version 0.59.0 (14 December 2023)
```

File: docs/source/release/0.59.0-notes.rst [hunk 3]
```diff
@@ -0,0 +1,527 @@
+
+Version 0.59.0 (14 December 2023)
+-----------------------------
```
Comment (stuartarchibald) [inline docs/source/release/0.59.0-notes.rst:3]:
```suggestion
Version 0.59.0rc1 (14 December 2023)
------------------------------------
```

File: docs/source/release/0.59.0-notes.rst [hunk 4]
```diff
@@ -0,0 +1,537 @@
...
+* PR `#9352 <https://github.com/numba/numba/pull/9352>`_: Removing object mode fallback from `@jit`. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#9353 <https://github.com/numba/numba/pull/9353>`_: Remove `numba.generated_jit` (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#9356 <https://github.com/numba/numba/pull/9356>`_: Refactor print tests to avoid NRT leak issue. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#9357 <https://github.com/numba/numba/pull/9357>`_: Fix a typo in `_set_init_process_lock` warning. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#9358 <https://github.com/numba/numba/pull/9358>`_: Remove note about OpenMP restriction in wheels. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#9359 <https://github.com/numba/numba/pull/9359>`_: Fix test_jit_module test against objmode fallback. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#9360 <https://github.com/numba/numba/pull/9360>`_: AzureCI changes. RVSDG test config should still test its assigned test slice (`sklam <https://github.com/sklam>`_)
```
Comment (stuartarchibald) [inline docs/source/release/0.59.0-notes.rst:515]:
```suggestion
* PR `#9360 <https://github.com/numba/numba/pull/9360>`_: AzureCI changes. RVSDG test config should still test its assigned test slice (`sklam <https://github.com/sklam>`_)
* PR `#9362 <https://github.com/numba/numba/pull/9362>`_: Fix ``np.MachAr`` warning matching in test. (`sklam <https://github.com/sklam>`_)
```
Manual addition of the unmerged #9362, this PR will be needed on `main` prior to branching for the 0.59 series of Numba.

File: docs/source/release/0.59.0-notes.rst [hunk 5]
```diff
@@ -0,0 +1,537 @@
...
+=======================================
+
+Support is added for ``numpy.unwrap()``. The ``axis`` argument is only supported when its value equals -1.
+
+(`PR-#9154 <https://github.com/numba/numba/pull/9154>`__)
+
+Adds support for checking if dtypes are equal.
+==============================================
+
+Support is added for checking if two dtype objects are equal, for example `assert X.dtype == np.dtype(np.float64)`.
```
Comment (stuartarchibald) [inline docs/source/release/0.59.0-notes.rst:122]:
```suggestion
Support is added for checking if two dtype objects are equal, for example ``assert X.dtype == np.dtype(np.float64)``.
```
two backticks in RST.

File: docs/source/release/0.59.0-notes.rst [hunk 6]
```diff
@@ -0,0 +1,537 @@
...
+
+This PR fixes `Numba #9058 <https://github.com/numba/numba/pull/9058>`_
+where it is now possible to call a guvectorize with multiple outputs.
+
+(`PR-#9049 <https://github.com/numba/numba/pull/9049>`__)
+
+Handling of `None` args fixed in `PythonAPI.call`.
+==================================================
+
+Fixing segfault when `args=None` was passed to `PythonAPI.call`.
```
Comment (stuartarchibald) [inline docs/source/release/0.59.0-notes.rst:210]:
```suggestion
Fixing segfault when ``args=None`` was passed to ``PythonAPI.call``.
```

File: docs/source/release/0.59.0-notes.rst [hunk 7]
```diff
@@ -0,0 +1,537 @@
...
+Support multiple outputs in a ``@guvectorize`` function
+=======================================================
+
+This PR fixes `Numba #9058 <https://github.com/numba/numba/pull/9058>`_
+where it is now possible to call a guvectorize with multiple outputs.
+
+(`PR-#9049 <https://github.com/numba/numba/pull/9049>`__)
+
+Handling of `None` args fixed in `PythonAPI.call`.
+==================================================
```
Comment (stuartarchibald) [inline docs/source/release/0.59.0-notes.rst:208]:
```suggestion
Handling of ``None`` args fixed in ``PythonAPI.call``.
======================================================
```

**Review by stuartarchibald — COMMENTED** [review]
Suggestions are added for applying two back-ticks as needed in the actual release note text so as to get it to render more suitably, I've left the list of PRs verbatim, it suffers from the same but is less likely to be read.

File: docs/source/release/0.59.0-notes.rst [hunk 8]
```diff
@@ -0,0 +1,537 @@
...
+* `kkokkoros <https://github.com/kkokkoros>`_
+* `KrisMinchev <https://github.com/KrisMinchev>`_
+* `munahaf <https://github.com/munahaf>`_
+* `njriasan <https://github.com/njriasan>`_
+* `rjenc29 <https://github.com/rjenc29>`_
+* `saulshanabrook <https://github.com/saulshanabrook>`_
+* `sklam <https://github.com/sklam>`_
+* `SridharCR <https://github.com/SridharCR>`_
+* `stuartarchibald <https://github.com/stuartarchibald>`_
+* `Tialo <https://github.com/Tialo>`_
```
Comment (stuartarchibald) [inline docs/source/release/0.59.0-notes.rst:537]:
```suggestion
```
Remove as https://github.com/numba/numba/blob/d4460feb8c91213e7b89f97b632d19e34a776cd3/docs/source/release/0.58.1-notes.rst#L33 was in 0.58.1

File: docs/source/release/0.59.0-notes.rst [hunk 9]
```diff
@@ -0,0 +1,537 @@
...
+* PR `#9173 <https://github.com/numba/numba/pull/9173>`_: Towncrier fixups (Continue #9158 and retarget to main branch) (`sklam <https://github.com/sklam>`_ `esc <https://github.com/esc>`_)
+* PR `#9181 <https://github.com/numba/numba/pull/9181>`_: Remove extra decrefs in RNG (`sklam <https://github.com/sklam>`_)
+* PR `#9189 <https://github.com/numba/numba/pull/9189>`_: Align `searchsorted` behaviour with numpy (`rjenc29 <https://github.com/rjenc29>`_)
+* PR `#9190 <https://github.com/numba/numba/pull/9190>`_: Fix issue with incompatible multiprocessing context in test. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#9191 <https://github.com/numba/numba/pull/9191>`_: Add a Numba power-on-self-test script and use in CI. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#9205 <https://github.com/numba/numba/pull/9205>`_: release notes and version support updates from release0.58 branch (`esc <https://github.com/esc>`_)
+* PR `#9220 <https://github.com/numba/numba/pull/9220>`_: Support passing arbitrary flags to NVVM (`gmarkall <https://github.com/gmarkall>`_)
```
Comment (stuartarchibald) [inline docs/source/release/0.59.0-notes.rst:471]:
```suggestion
```
This was in 0.58.1. Needs removing.
https://github.com/numba/numba/blob/d4460feb8c91213e7b89f97b632d19e34a776cd3/docs/source/release/0.58.1-notes.rst#L32

File: docs/source/release/0.59.0-notes.rst [hunk 10]
```diff
@@ -0,0 +1,537 @@
...
+* PR `#9159 <https://github.com/numba/numba/pull/9159>`_: Fix uncaught exception in `find_file()` (`gmarkall <https://github.com/gmarkall>`_)
+* PR `#9168 <https://github.com/numba/numba/pull/9168>`_: fix the `get_template_info` method in `overload_method` template (`sklam <https://github.com/sklam>`_)
+* PR `#9169 <https://github.com/numba/numba/pull/9169>`_: Update `np.digitize` handling of np.nan bin edge(s) (`rjenc29 <https://github.com/rjenc29>`_)
+* PR `#9170 <https://github.com/numba/numba/pull/9170>`_: Fix an inappropriate test expression to remove a logical short circuit (`munahaf <https://github.com/munahaf>`_)
+* PR `#9171 <https://github.com/numba/numba/pull/9171>`_: Fix the implementation of a special method (`munahaf <https://github.com/munahaf>`_)
+* PR `#9173 <https://github.com/numba/numba/pull/9173>`_: Towncrier fixups (Continue #9158 and retarget to main branch) (`sklam <https://github.com/sklam>`_ `esc <https://github.com/esc>`_)
+* PR `#9181 <https://github.com/numba/numba/pull/9181>`_: Remove extra decrefs in RNG (`sklam <https://github.com/sklam>`_)
```
Comment (stuartarchibald) [inline docs/source/release/0.59.0-notes.rst:466]:
```suggestion
```
This was in 0.58.0. Needs removing.
https://github.com/numba/numba/blob/d4460feb8c91213e7b89f97b632d19e34a776cd3/docs/source/release/0.58.0-notes.rst#L522

File: docs/source/release/0.59.0-notes.rst [hunk 11]
```diff
@@ -0,0 +1,537 @@
...
+* PR `#9155 <https://github.com/numba/numba/pull/9155>`_: Remove unused test.cmd (`sklam <https://github.com/sklam>`_)
+* PR `#9159 <https://github.com/numba/numba/pull/9159>`_: Fix uncaught exception in `find_file()` (`gmarkall <https://github.com/gmarkall>`_)
+* PR `#9168 <https://github.com/numba/numba/pull/9168>`_: fix the `get_template_info` method in `overload_method` template (`sklam <https://github.com/sklam>`_)
+* PR `#9169 <https://github.com/numba/numba/pull/9169>`_: Update `np.digitize` handling of np.nan bin edge(s) (`rjenc29 <https://github.com/rjenc29>`_)
+* PR `#9170 <https://github.com/numba/numba/pull/9170>`_: Fix an inappropriate test expression to remove a logical short circuit (`munahaf <https://github.com/munahaf>`_)
+* PR `#9171 <https://github.com/numba/numba/pull/9171>`_: Fix the implementation of a special method (`munahaf <https://github.com/munahaf>`_)
+* PR `#9173 <https://github.com/numba/numba/pull/9173>`_: Towncrier fixups (Continue #9158 and retarget to main branch) (`sklam <https://github.com/sklam>`_ `esc <https://github.com/esc>`_)
```
Comment (stuartarchibald) [inline docs/source/release/0.59.0-notes.rst:465]:
```suggestion
```
This was in 0.58.0. Needs removing.
https://github.com/numba/numba/blob/d4460feb8c91213e7b89f97b632d19e34a776cd3/docs/source/release/0.58.0-notes.rst#L521

File: docs/source/release/0.59.0-notes.rst [hunk 12]
```diff
@@ -0,0 +1,537 @@
...
+* PR `#9142 <https://github.com/numba/numba/pull/9142>`_: Removed out of date comment handled by PR#8338 (`njriasan <https://github.com/njriasan>`_)
+* PR `#9144 <https://github.com/numba/numba/pull/9144>`_: Fix error when literal is wrongly propagated in a PHI node (`guilhermeleobas <https://github.com/guilhermeleobas>`_)
+* PR `#9148 <https://github.com/numba/numba/pull/9148>`_: bump llvmdev dependency to 0.42.0dev for next development cycle (`esc <https://github.com/esc>`_)
+* PR `#9149 <https://github.com/numba/numba/pull/9149>`_: update release checklist post 0.58.0rc1 (`esc <https://github.com/esc>`_)
+* PR `#9152 <https://github.com/numba/numba/pull/9152>`_: Fix old_style error capturing deprecation warnings (`sklam <https://github.com/sklam>`_)
+* PR `#9154 <https://github.com/numba/numba/pull/9154>`_: Add support for np.unwrap() (`KrisMinchev <https://github.com/KrisMinchev>`_)
+* PR `#9155 <https://github.com/numba/numba/pull/9155>`_: Remove unused test.cmd (`sklam <https://github.com/sklam>`_)
+* PR `#9159 <https://github.com/numba/numba/pull/9159>`_: Fix uncaught exception in `find_file()` (`gmarkall <https://github.com/gmarkall>`_)
```
Comment (stuartarchibald) [inline docs/source/release/0.59.0-notes.rst:460]:
```suggestion
```
This was in 0.58.0. Needs removing.
https://github.com/numba/numba/blob/d4460feb8c91213e7b89f97b632d19e34a776cd3/docs/source/release/0.58.0-notes.rst#L520

File: docs/source/release/0.59.0-notes.rst [hunk 13]
```diff
@@ -0,0 +1,537 @@
...
+* PR `#9189 <https://github.com/numba/numba/pull/9189>`_: Align `searchsorted` behaviour with numpy (`rjenc29 <https://github.com/rjenc29>`_)
+* PR `#9190 <https://github.com/numba/numba/pull/9190>`_: Fix issue with incompatible multiprocessing context in test. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#9191 <https://github.com/numba/numba/pull/9191>`_: Add a Numba power-on-self-test script and use in CI. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#9205 <https://github.com/numba/numba/pull/9205>`_: release notes and version support updates from release0.58 branch (`esc <https://github.com/esc>`_)
+* PR `#9220 <https://github.com/numba/numba/pull/9220>`_: Support passing arbitrary flags to NVVM (`gmarkall <https://github.com/gmarkall>`_)
+* PR `#9223 <https://github.com/numba/numba/pull/9223>`_: CUDA: Add support for compiling device functions with C ABI (`gmarkall <https://github.com/gmarkall>`_)
+* PR `#9227 <https://github.com/numba/numba/pull/9227>`_: Support NumPy 1.26 (PR aimed at review / merge) (`gmarkall <https://github.com/gmarkall>`_ `Tialo <https://github.com/Tialo>`_)
```
Comment (stuartarchibald) [inline docs/source/release/0.59.0-notes.rst:473]:
```suggestion
```
This was in 0.58.1. Needs removing.
https://github.com/numba/numba/blob/d4460feb8c91213e7b89f97b632d19e34a776cd3/docs/source/release/0.58.1-notes.rst#L33

Github username `Tialo` needs removing from `Authors` below as a consequence of removing this entry.

File: docs/source/release/0.59.0-notes.rst [hunk 14]
```diff
@@ -0,0 +1,537 @@
...
+* PR `#9228 <https://github.com/numba/numba/pull/9228>`_: Fix #9222 - Don't replace `.` with `_` in func arg names in inline closures (`gmarkall <https://github.com/gmarkall>`_)
```
Comment (stuartarchibald) [inline docs/source/release/0.59.0-notes.rst:474]:
```suggestion
```
This was in 0.58.1. Needs removing.
https://github.com/numba/numba/blob/d4460feb8c91213e7b89f97b632d19e34a776cd3/docs/source/release/0.58.1-notes.rst#L34

File: docs/source/release/0.59.0-notes.rst [hunk 15]
```diff
@@ -0,0 +1,537 @@
...
+* PR `#9169 <https://github.com/numba/numba/pull/9169>`_: Update `np.digitize` handling of np.nan bin edge(s) (`rjenc29 <https://github.com/rjenc29>`_)
+* PR `#9170 <https://github.com/numba/numba/pull/9170>`_: Fix an inappropriate test expression to remove a logical short circuit (`munahaf <https://github.com/munahaf>`_)
+* PR `#9171 <https://github.com/numba/numba/pull/9171>`_: Fix the implementation of a special method (`munahaf <https://github.com/munahaf>`_)
+* PR `#9173 <https://github.com/numba/numba/pull/9173>`_: Towncrier fixups (Continue #9158 and retarget to main branch) (`sklam <https://github.com/sklam>`_ `esc <https://github.com/esc>`_)
+* PR `#9181 <https://github.com/numba/numba/pull/9181>`_: Remove extra decrefs in RNG (`sklam <https://github.com/sklam>`_)
+* PR `#9189 <https://github.com/numba/numba/pull/9189>`_: Align `searchsorted` behaviour with numpy (`rjenc29 <https://github.com/rjenc29>`_)
+* PR `#9190 <https://github.com/numba/numba/pull/9190>`_: Fix issue with incompatible multiprocessing context in test. (`stuartarchibald <https://github.com/stuartarchibald>`_)
```
Comment (stuartarchibald) [inline docs/source/release/0.59.0-notes.rst:468]:
```suggestion
```
This was in 0.58.0. Needs removing.
https://github.com/numba/numba/blob/d4460feb8c91213e7b89f97b632d19e34a776cd3/docs/source/release/0.58.0-notes.rst#L523

File: docs/source/release/0.59.0-notes.rst [hunk 16]
```diff
@@ -0,0 +1,537 @@
...
+* PR `#9140 <https://github.com/numba/numba/pull/9140>`_: Add support for Polynomial class (`KrisMinchev <https://github.com/KrisMinchev>`_)
+* PR `#9141 <https://github.com/numba/numba/pull/9141>`_: Add support for `as_series()` from `np.polynomial.polyutils` and `polydiv()`, `polyint()`, `polyval()` from `np.polynomial.polynomial` (`KrisMinchev <https://github.com/KrisMinchev>`_)
+* PR `#9142 <https://github.com/numba/numba/pull/9142>`_: Removed out of date comment handled by PR#8338 (`njriasan <https://github.com/njriasan>`_)
+* PR `#9144 <https://github.com/numba/numba/pull/9144>`_: Fix error when literal is wrongly propagated in a PHI node (`guilhermeleobas <https://github.com/guilhermeleobas>`_)
+* PR `#9148 <https://github.com/numba/numba/pull/9148>`_: bump llvmdev dependency to 0.42.0dev for next development cycle (`esc <https://github.com/esc>`_)
+* PR `#9149 <https://github.com/numba/numba/pull/9149>`_: update release checklist post 0.58.0rc1 (`esc <https://github.com/esc>`_)
+* PR `#9152 <https://github.com/numba/numba/pull/9152>`_: Fix old_style error capturing deprecation warnings (`sklam <https://github.com/sklam>`_)
```
Comment (stuartarchibald) [inline docs/source/release/0.59.0-notes.rst:457]:
```suggestion
```
This was in 0.58.0. Needs removing.
https://github.com/numba/numba/blob/d4460feb8c91213e7b89f97b632d19e34a776cd3/docs/source/release/0.58.0-notes.rst#L519

**Review by stuartarchibald — COMMENTED** [review]
This review focuses on removing duplicate entries present in previous releases.

**Review by stuartarchibald — APPROVED** [review]
Thanks for the patch and fixes!

**Pull request merged**